### ISSUE TITLE
VIC 1.1 Flex plug-in not installing on vSphere 6.5 

### ISSUE THREAD

**Issue by vmwarelab** [vmware/vic#4981]
**User Statement:**

VIC 1.1 Flex plug-in not installing on vSphere 6.5 

**Details:**
Installation details after configuring the configs file with the vCenter FQDN

====================================================
root@photon-xjYHNzlHB [ /workspace/vic/ui/VCSA ]# ./install.sh
Enter your vCenter Administrator Username: administrator@vsphere.local
Enter your vCenter Administrator Password:
----------------------------------------
Checking if VCSA has Bash shell enabled...
Please enter the root password
----------------------------------------
Warning: the ECDSA host key for 'vcs-l-01a.vmwarelab.local' differs from the key for the IP address '192.168.0.20'
Offending key for IP in /root/.ssh/known_hosts:1
Matching host key in /root/.ssh/known_hosts:2
Are you sure you want to continue connecting (yes/no)? yes
root@vcs-l-01a.vmwarelab.local's password:
failed to find target plugin (com.vmware.vicui.Vicui)
Reading plugin-package.xml...
./install.sh: line 125: ../vsphere-client-serenity/-/vc_extension_flags: No such file or directory
----------------------------------------
Registering vCenter Server Extension...
----------------------------------------
--name must be specified
-------------------------------------------------------------
Error! Could not register plugin with vCenter Server. Please see the message above
root@photon-xjYHNzlHB [ /workspace/vic/ui/VCSA ]#
============================================================

**VIC version: 1.1 **

**Steps to reproduce:**
Just installing the Flex plug-in within the VCSA folder after configuring the configs file with only the vCenter FQDN

**Actual behavior:**
Error! Could not register plugin with vCenter Server. Please see the message above

**Expected behavior:**
successful register the pluging with vCenter Server

**Logs:**
None

**Comment by vmwarelab** [comment]
what a bad experience. this is very frustrating that you cant install a simple plug-in.
found from various thread that we actually shipped a broken script "install.sh' when installing the flex plug-in which i managed to workaround it by copying the install.sh from the HTML folder instead and modify the plugin_type so it = flex in the script file. 

now that the plug-in is finally installed i have another issues .. i see the plug-in i click on it and it just hangs processing with no data output. screen attached 
![image](https://cloud.githubusercontent.com/assets/19843322/25620537/c458d042-2f1c-11e7-9988-184b88e383e2.png)

**Comment by anchal-agrawal** [comment]
Ping @jooskim for triage.

**Comment by jooskim** [comment]
I'm sorry @vmwarelab to have you go through another pain. What I understand about this issue is that even after successful installation you won't be able to see the portlets in the VCH or Container VM's Summary page. I am very aware of this issue and it was addressed in the v1.1.1 release. Please see https://github.com/vmware/vic/pull/4951. If you are willing to install the updated plugin right now I suggest you download a build that has this fix (https://bintray.com/vmware/vic-repo/download_file?file_path=vic_10178.tar.gz)

And for your information, the UI plugin comes in two versions - Flex client version and H5 client version. I see that you installed the plugin in the Flex client which should NOT work the way you expect. The plugin for the Flex client should ONLY display the portlets mentioned above. That being said, the fact you could navigate into the vSphere Integrated Containers section in the Flex client seems like a bug, as it was not supposed to be shown there.

**Comment by jooskim** [comment]
@vmwarelab The broken install.sh script bug is also addressed in the same PR above.

**Comment by jooskim** [comment]
Also, when reinstalling the plugin make sure to restart the Flex client after successful deployment.

**Comment by vmwarelab** [comment]
thank you @jooskim i will uninstall the plug-in and test the new build 
thank you for your help as always 

any idea when this will be released ?

**Comment by jooskim** [comment]
@vmwarelab Actually you can download this version https://bintray.com/vmware/vic/Download/v1.1.1-rc1. I don't pretty much have an idea on that. Pinging @mdubya66.

**Comment by jooskim** [comment]
I just talked with our manager Matt and here's what I've got. v1.1.1 is now in final QA but RC2 is in the plan for Admiral-related bugfixes. For the OVA file you could get it [here](https://storage.googleapis.com/vic-product-ova-dev-builds/vic-v1.1.1-rc1-a1f00c54.ova)

**Comment by vmwarelab** [comment]
Awesome! thanks @jooskim Josh

**Comment by vmwarelab** [comment]
@jooskim Hi Josh!   do i need to set the VIC_UI_HOST_URL="NOURL" even though now it says deprecated , use it on your own risk!!   ? since i m not using the the VIC appliance for the plug-in installation

**Comment by mdubya66** [comment]
Can we close this a duplicate?

**Comment by vmwarelab** [comment]
@mdubya66 Hey Matt. i m still looking for some answers based on my last post.. thank you

**Comment by jooskim** [comment]
@vmwarelab It is failing as expected, as we no longer ship the plugin zip file as we used to with the vic engine tar ball file. The reason behind it is that the previous approach was strongly discouraged during our internal review. Still, the install.sh file itself is capable of doing that. It's just that you have to put the plugin zip file (which you could find inside the OVA appliance) under `vic/ui/plugin-packages/com.vmware.vic-v1.1.1.zip` 

Please be aware that we have to yet document the procedure way to install the plugin using the deprecated method (NOURL), so if you still plan to install the plugin this way please first let me know what your `plugin-manifest` file looks like which can be found at `vic/ui/plugin-manifest`

**Comment by vmwarelab** [comment]
@jooskim i just need clear steps Josh.. i feel i m shooting in the dark here and guessing. 

I only downloaded the vic_1.1.1-rc1.tar.gz  and extracted on a photon machine where i m executing everything from .  deploying the vCH works fine but i m still struggling with the both, the flex and html5 plug-in installation . 

Steps i m taking right now and please modify / correct it if i m doing it wrong 

1. Modify the configs file to add the vCenter IP 
2. Setting VIC_UI_HOST_URL to VIC appliance i deployed "vic-v1.1.0-bf760ea2" but i didn't copy any zip files since you never told me i need those files , where can i get these zip files  ? is there more than one ?
3. Setting VIC_UI_HOST_THUMBPRINT from the VIC appliance 
4. modify the plugin_type in the configs file  to flex and run the install.sh script
5. repeat step 4 for plugin_type html5 and run the install.sh script again

so far nothing showed up in any of the clients even though the installation was successful. but no vSphere integrated Container Icon/object created.

**Comment by jooskim** [comment]
@vmwarelab Oh so you did deploy the OVA. Then you simply have to set the VIC_UI_HOST_URL variable to `https://${IP_ADDRESS_OF_OVA_APPLIANCE}:9443/` (type in this URL in the browser to see if you can connect to this file server) The critical piece of information i didn't tell you is that the OVA already has all the plugin zip files both for the Flex plugin and H5 plugin. I admit that it's not clearly documented in any part of our documentation, so I will make sure with our documentation team that the steps are clearly indicated. I think there's actually ongoing work but I will double check.

**Comment by jooskim** [comment]
Again, sorry for the bad experience.

**Comment by vmwarelab** [comment]
@jooskim no worries lol.. i m learning. okay the files on the ova are 1.1 not 1.1.1 so i moved the 1.1 zip files into a folder and i copied the 1.1.1 from the vic/ui/plugin-packages/com.vmware.vic-v1.1.1.zip after extracting the 1.1.1-rc1 binary ..  will that be okay ?

**Comment by vmwarelab** [comment]
i did the installation twice .. once for the flex and once for the html5 and both were successful but not functional 

flex screen shots

![flex1](https://cloud.githubusercontent.com/assets/19843322/25645735/4becc34e-2f7f-11e7-8c83-98981f8a6a0e.png)
![flex2](https://cloud.githubusercontent.com/assets/19843322/25645736/4c0497d0-2f7f-11e7-9d2b-b89437eead33.png)
![flex3](https://cloud.githubusercontent.com/assets/19843322/25645737/4c09126a-2f7f-11e7-8cd3-97bab47181aa.png)

html screen shots

![html1](https://cloud.githubusercontent.com/assets/19843322/25645743/55ace67a-2f7f-11e7-8303-bd51e34c1342.png)
![html2](https://cloud.githubusercontent.com/assets/19843322/25645742/55a3f448-2f7f-11e7-9b80-f7b8bc68a6ea.png)

**Comment by jooskim** [comment]
So here are two options for you. Choose either way, and remember all these are not an official method to install the plugin. Our documentation team is currently working on it so stay tuned.

1. Using the OVA appliance
- Open the `vic/ui/plugin-manifest` file and make sure that `version` is set to `1.1.1`.
- Copy the plugin zip file `com.vmware.vic-v1.1.1.zip` to the OVA appliance by using `scp` to `/data/fileserver/files/` in the remote host
- Make sure VIC_UI_HOST_URL is set to `https://${OVA_APPLIANCE_IP}:9443/` and VIC_UI_HOST_THUMBPRINT is set
- Run install.sh
- You may have to restart the vSphere Client or vSphere Web Client after successful deployment

2. `vic/ui/plugin-packages/~~~.zip` is ONLY necessary if you are going with the NOURL approach, but in that case make sure that:
- VIC_UI_HOST_URL is set to "NOURL"
- The version variable in the `plugin-manifest` file is set to `1.1.1`
- run install.sh
- You may have to restart the vSphere Client or vSphere Web Client after successful deployment

**Comment by vmwarelab** [comment]
lets go with the first approach 

the manifest is as follows : 

root@photon-xjYHNzlHB [ /workspace/vic/ui ]# vi plugin-manifest
key_flex="com.vmware.vic.ui"
name="VicUI"
version="1.1.1-rc1"  ---> are you saying release this with 1.1.1 ?
summary="VicUI"
company="VMware"
key_h5c="com.vmware.vic"

i did copy the com.vmware.vic-v1.1.1.zip but its actually called com.vmware.vic-v1.1.1-rc1.zip  do i need to rename it ?

![vic](https://cloud.githubusercontent.com/assets/19843322/25645831/5d4d0558-2f80-11e7-8064-c5b18277d306.png)

**Comment by jooskim** [comment]
Thanks for posting the screenshot. This makes it easier for me to what to edit. First let's rename `version="1.1.1-rc1"` to `version="1.1.1"`, as the vSphere Client will throw out an error when it sees extra strings like "-rc1", which means you also have to rename the `com.vmware.vic-v1.1.1-rc1.zip` to `com.vmware.vic-v1.1.1.zip`

**Comment by vmwarelab** [comment]
@jooskim okay we are getting somewhere.. the Icon shows now in both type of clients (flex & html )
but

issues i m seeing with the flex client 

1. no VCH portlets that i can identify for the VCH endpoint
![flex](https://cloud.githubusercontent.com/assets/19843322/25646186/e06098c6-2f83-11e7-8f89-6887476c90d3.png)

2. when clicking the vSphere Integrated container Icon .. we sit in a continues loop with no data showing 

![flex4](https://cloud.githubusercontent.com/assets/19843322/25646178/c5663922-2f83-11e7-91c3-05ca87362b99.png)

for the html ..seem to be working but it only gives me a count of VCH and Containers deployed.

**Comment by jooskim** [comment]
The `vSphere Integrated Containers` section is not supposed to be shown in the Flex Client, but you are clearly seeing that. This definitely seems like a bug and I need to investigate how and why this happened. Give me some time to reproduce it tomorrow. It is possible, if you remember this issue https://github.com/vmware/vic/issues/3089, that you might be hitting this issue https://ikb.vmware.com/kb/2109074. To be sure that's the case could you look into the `/var/log/vmware/vsphere-client/logs/vsphere_client_virgo.log` in the VCSA appliance, and it contains something like the following?
```
[2017-04-12T21:07:09.404Z] [ERROR] data-service-pool-1365       70000297 100012 200006 com.vmware.vicui.VicUIServiceImpl                                 VicUIServiceImpl.getProperties error: com.sun.xml.internal.ws.client.ClientTransport
Exception: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
```

**Comment by jooskim** [comment]
For the HTML5 Client what you're seeing is what's intended for v1.1.0. I know there really isn't much of a point in displaying how many VCHs are there, but more features are currently in development for the next release (1.2)

**Comment by vmwarelab** [comment]
@jooskim will check the logs and let you know

**Comment by vmwarelab** [comment]
@jooskim this is the only warning i m seeing 

[2017-05-03T02:28:28.293Z] [WARN ] data-service-pool-798        70000373 100007 200006 com.vmware.vic.VicUIDataAdapter                                   CompositeConstraint is unsupported

**Comment by jooskim** [comment]
In the H5 Client Summary page for the VCH VM scroll down enough. Do you still not see the Virtual Container Host portlet? In that case, did you make sure `VCH01` was deployed using the latest build of VIC Engine? Some backend changes were made to the VCH appliance specs quite a bit of time ago which eventually broke the Flex client plugin (which is addressed in v1.1.1). Also please look into `/var/log/vmware/vsphere-ui/logs/vsphere_client_virgo.log` and search for any [ERROR] instance that has text `com.vmware.vic`.

By the way, I know you probably might be aware of this but just in case you don't, the logs file for the Flex and H5 Clients can be found in the respective locations:

1. Flex: `/var/log/vmware/vsphere-client/logs/vsphere_client_virgo.log`
2. H5C: `/var/log/vmware/vsphere-ui/logs/vsphere_client_virgo.log`

**Comment by vmwarelab** [comment]
@jooskim 

i do not see the VCH portlet , i sent you a full page capture 
VCH01 is using 1.1.1-rc1 "Photon - VCH v1.1.1-rc1, 10186, a1f00c5" and the vAPP version is v1.1.1-rc1  as well. 

no error with anything that has com.vmware.vic.. only the warning i mentioned earlier..

i ll leave it to you for now. i m calling it a day .. thank you so much for your continuous support and help

**Comment by jooskim** [comment]
@vmwarelab I'm back now. Let me first figure out some basic information on your vSphere setup. If none of my guesses worked out then it's probable that something is conflicting internally. I tried installing the plugin on vSphere 6.5.0 and 6.5.0d (latest public version) instances and it worked as expected in the both cases.

Could you let me know which version of vSphere (including build number) you are using (and if you have upgraded from which version)?

**Comment by vmwarelab** [comment]
@jooskim 

Yes it was an actually an upgrade from 6.0 - 6.5

Product:
VMware vCenter Server Appliance
Version: 6.5.0.5500

**Comment by jooskim** [comment]
@vmwarelab The latest public version I could find is 6.5.0.5300 at https://my.vmware.com/web/vmware/details?downloadGroup=ESXI650D&productId=614&rPId=15967. Could you point me to the place where you proceeded with your upgrade?

**Comment by vmwarelab** [comment]
Yes but the esxi host version . What i provided was yhe vcenter version

**Comment by jooskim** [comment]
Whatever the VCSA version it might be, it seems to me that the upgrade process somehow screwed up the plugin. Is your VCSA instance in a private network? I thought I could probably SSH into your VCSA host and see what's going on, if it's in public network. Also could you tell me how you upgraded your older version of VCSA 6.0 to 6.5? What I could do is take the exact same steps you went through.

**Comment by vmwarelab** [comment]
@jooskim Hi Josh.. sure let me know when you want to do that and i will email you internally 
the details on how you can ssh to the VCSA

**Comment by jooskim** [comment]
@vmwarelab Sorry it took a while for me to come back. I was on PTO last week. You can send me details anytime you feel comfortable. Thanks.

**Comment by Luckeyd** [comment]
I have been following this thread for a few days as I am seeing the exact same outcome @vmwarelab is.  I also did an upgrade/migration from VCSA 6.0 to 6.5.  My results are exactly like @vmwarelab.  

For reference my VCSA is versions are:

VMware vCenter Server Appliance 6.5.0.5500 Build Number 5318154
vSphere Web Client 6.5.0 Build 5318154.

The version I am running came directly from the latest ISO downloaded from my.vmware.com then going to VCSA-IP:5480 and running the "Update" option.

**Comment by jooskim** [comment]
@vmwarelab @Luckeyd Thanks for sharing details with me. What I am going to do is install a VCSA 6.0 instance and then upgrade it to the latest public version 6.5.0d (VCVA build 5318154) and see if I can reproduce the problem. Stay tuned.

**Comment by jooskim** [comment]
@vmwarelab @Luckyed After installing a VCSA 6.0.2 instance and upgrading it to 6.5.0d, I still see the plugin deploying and working fine. It seems to me that there could have been something wrong in the installation procedure you followed. For me able to further assess the issue I would either need remote access to your host or we could do a screencast.

**Comment by jooskim** [comment]
@vmwarelab @Luckeyd Hey all, I think I might have found a solution to this issue. Assuming you have all successfully deployed the HTML5 Client plugin but NOT Flex Client plugin, then issue #5151 could be the one you are experiencing. Let me know when you are available. I'll send you a build with the fix and let's see if that fixes your problem.

**Comment by vmwarelab** [comment]
hi @jooskim I can give it a shot on the weekend if you want to send me the new build. just busy these couple of days

**Comment by jooskim** [comment]
@vmwarelab it's currently in full ci build and will take a couple hours from now. will keep you updated :)

**Comment by jooskim** [comment]
@vmwarelab @Luckeyd Here's [the link](https://storage.googleapis.com/vic-engine-builds/vic_10673.tar.gz) to the build 10673 that includes the fix. Unfortunately I just got the PR merged for this fix yesterday so it would be available in the official 1.1.1 patch, but I would make sure those are included in the next patch release.

**Comment by vmwarelab** [comment]
@jooskim i will test it and let you know as soon as i can

**Comment by jooskim** [comment]
@vmwarelab any updates you could provide at this moment?

**Comment by jooskim** [comment]
Hi @vmwarelab, just wanted to check back on this issue to see if you had a chance to try out the fix. Once you can confirm it's fixed I can close the issue.

**Comment by Luckeyd** [comment]
Sorry for the delays on my end, I have been traveling the last few weeks.  I just downloaded build 10673 and will be giving it a try this week,

**Comment by vmwarelab** [comment]
@jooskim Sorry Josh . been a busy month . and i m going on PTO soon.. 
i have installed both the html5 and flex using v1.1.1-rc1  .. this is the code you wanted us to test correct ?

for me i dont see any change, so i still do not see the VCH portlet in the flex client . 
again my environment gone though many upgrades since 5x release to 6.5 now

**Comment by jooskim** [comment]
@vmwarelab Good to see you again. So have you downloaded `vic_10673.tar.gz`? One question I have to check if the plugin has been deployed and enabled successfully would be to check the home screen of the Flex Client. Do you still see shortcut for vSphere Integrated Containers? Also, did you restart the vSphere Flex Client after deploying the the Flex plugin?

**Comment by vmwarelab** [comment]
yes i did .. and i do see the short cut in both clients types.. and i did restart both services.

**Comment by jooskim** [comment]
@vmwarelab That's a good sign to me, at least for the debug purpose. As of build 10673, you should not be seeing the shortcut in the Flex client. This leads me to think that the plugin was not deployed properly. Is your VC accessible publicly or is it in private network?

**Comment by vmwarelab** [comment]
i can do a webex and grant you access .. but currently on the phone.
send me an email internally and i will send you the invite

**Comment by jooskim** [comment]
I don't know your email unfortunately. Do you have a Slack account with vmware.slack.com? If you do find me at `j_kim`

**Comment by vmwarelab** [comment]
malasfar@vmware.com

**Comment by Luckeyd** [comment]
I downloaded the 10673 update and I am also not seeing a change in my environment.  No shortcut in Flex client.

**Comment by jooskim** [comment]
@Luckeyd I had a call with @vmwarelab yesterday where his VC setup had an issue with certificates. Did you also upgrade your VC from 6.0 to 6.5? If that's the case, could locate the vSphere Web Client log file at `/var/log/vmware/vsphere-client/logs/vsphere_client_virgo.log` on your VCSA host, open it in any editor, and look for a message that looks like the following?

```
Exception: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
```

**Comment by Luckeyd** [comment]
I have had success on my install for the HTML5 client.

![icon-html5](https://user-images.githubusercontent.com/6510300/27241138-63169d08-528c-11e7-905c-c89c28458fbd.PNG)

![details-html5](https://user-images.githubusercontent.com/6510300/27241141-66acc596-528c-11e7-8866-310eeb50ebc6.PNG)

**Comment by RobTiki** [comment]
@jooskim 
I am currently having this issue. 
`Exception: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match`
I checked the [KB2121689](https://kb.vmware.com/selfservice/search.do?cmd=displayKC&docType=kc&docTypeID=DT_KB_1_1&externalId=2121689) and my vcsa certs match and looks ok.

I have tried VIC 1.1.1 and 1.1.1-rc2.  

The problem seems to be on the VIC engine vm.  When accessing vic_ova:9443 I checked the cert and it was issued to :9443 by :9443.

Let me know if you need any additional logs.  The error is the same as the OP and

**Comment by jooskim** [comment]
@Luckeyd @RobTiki Sorry, I have been busy with other high priority stuff. This will be one of my primary focuses for this and next weeks. I'll follow up with each of you for detailed troubleshooting soon.

**Comment by jooskim** [comment]
@Luckeyd That you are not seeing the shortcut for vSphere Integrated Containers is a good sign to me. It is not supposed to be present in the home screen, because that feature is not supported in the Flex Client. BUT you still should see "Virtual Container Host" portlet for a Virtual Container Host appliance [like this](https://user-images.githubusercontent.com/3834071/26953756-e64b61c0-4c71-11e7-98c3-9242ba25a7ab.png) and "Container" portlet for a Container VM

**Comment by RobTiki** [comment]
@jooskim I have done a fresh install of VIC 1.1.1 via the OVA and have the same issue.  No portlet on the vch but the vSphere Integrated Containers link shows on both the flex and h5 client home page.

I have tried setting the same certs that are on the ova registry/managment for the fileserver and adding them to the vcsa trusted store.  I am at a point where I don't know what else to try.  I am going to provide a lot of info hoping that you can help me resolve this.

When I run openssl from vcsa to the the fileserver on 9443 I get below:
```
 root@vCSA [ ~/vic/ui/VCSA ]# echo | opens
CONNECTED(00000003)                      
depth=0 O = "VMware, Inc.", CN = :9443   
verify error:num=20:unable to get local i
verify return:1                          
depth=0 O = "VMware, Inc.", CN = :9443   
verify error:num=21:unable to verify the 
verify return:1                          
---                                      
Certificate chain                        
 0 s:/O=VMware, Inc./CN=:9443            
   i:/O=VMware, Inc./CN=:9443            
---                                      
!!TRUNCATED BY ROB!!!
Start Time: 1498137913
    Timeout   : 300 (sec)
    Verify return code: 21 (unable to verify the first certificate)
---
DONE
```
I am also still getting:
`[2017-06-22T15:49:31.655Z] [ERROR] data-service-pool-788        70001518 100020 200002 com.vmware.vic.VicUIPropertyProvider                              VicUIServiceImpl.getProperties error: com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
`
Some screenshots below:

![image](https://user-images.githubusercontent.com/12755848/27442646-c7e26008-573e-11e7-8dfc-20d2d7a80156.png)

![image](https://user-images.githubusercontent.com/12755848/27442844-5cc047b2-573f-11e7-925a-34fec92d4ddb.png)

![image](https://user-images.githubusercontent.com/12755848/27442878-7992c5fe-573f-11e7-86b9-f41c950cff4c.png)

![image](https://user-images.githubusercontent.com/12755848/27443478-31fddd58-5741-11e7-94de-a191c6b956b2.png)

![image](https://user-images.githubusercontent.com/12755848/27443736-104477ca-5742-11e7-9043-7ecdc7d9f0d2.png)

**Comment by jooskim** [comment]
@RobTiki Thanks for the detailed report. You are affected by https://github.com/vmware/vic/pull/5154 which was addressed about a month ago. For the fix you can try the following OVA:
https://storage.googleapis.com/vic-product-ova-builds/vic-v1.1.1-11340-4029cbe6.ova

You don't need to deploy a new VCH again, just simply redeploy the OVA and reinstall the plugins (both H5 Client and Flex Client) based off of the new OVA appliance. Also don't forget to restart the both clients after installation is done.

Towards 1.2 release we are looking ways to simplify the UI plugin installation experience where the user will no longer have to edit the `configs` file, but for now can you try the new build?

**Comment by RobTiki** [comment]
@jooskim Gave it a shot. Uninstalled. Destroyed VIC ova and deployed the new build.  Installed plugins based on the [VIC documentation](https://vmware.github.io/vic-product/assets/files/html/1.1/vic_vsphere_admin/plugins_vcsa.html)  Same issue. 

Something in my vcsa server must be messed up.  Are there any details on what was done for the certs on @vmwarelab system?

I am at the point where I am going to give up on this since its not stopping me from exploring VIC and containers.

**Comment by jooskim** [comment]
@RobTiki So we came to this point where he said he would redo some of what he'd done before. I don't know exactly but it was something like reapplying the certificates. Maybe @vmwarelab can give you some hints here. By the way, are you still seeing the "vSphere Integrated Containers" shortcut on Flex Client even after reinstalling the plugin and restarting the vSphere Client and vSphere Web Client? I assume you ran `service-control --stop vsphere-client && service-control --start vsphere-client` and `service-control --stop vsphere-client && service-ui --start vsphere-ui`.

**Comment by jooskim** [comment]
@RobTiki I am going to do a deep log analysis after trying to replicate the issue by installing the official v1.1.1 release. Since I myself have not had the SSL handshake error, I will try to see if the issue #5154 is the culprit for the error. Stay tuned.

**Comment by RobTiki** [comment]
@jooskim It still shows on the home screen in h5 ui but not in flex vsphere-client after installing the new version.  vicUI is listed in Administration > Client Plug-in and vCenter Server Extensions in the flex client though. I tried just the h5 install, just the flex install and both. right now I am working just with the h5 vic plugin.

I think there is something going on with my vcsa certs.  I left the file server cert blank when deploying the ova. In your lab does the VIC-OVA cert for fileserver issued to :9443?

I need to get to the bottom of the cert thumbprint issue.  I tail the logs when I go to the vch vm in host and clusters view and the logs shows thats its trying to get the plugin properties but its erroring on the trust and thumbprint.  I'll go down that path.

@vmwarelab  or @Luckeyd please share how you resolved your issue.


*I like where the update is going with provision the VCH from a web interface. nice.

**Comment by RobTiki** [comment]
@jooskim is there a way for me to debug thumprintHostnameVerifier and thumbprintTrustManager classes.  I was looking into this again last night and cannot understand why it's failing. 

I created a self sign cert for the file server and added it to all the keystore and cacerts Store.  I also was able to use SSLPoke class to verify connectivity against each store file, total of 4 vsphere-ui, vsphere-client and VMware-virgo and the default java cacertstore.  I restated severed and also rebooted several times. 

I believe stepping through the trust validation would help find the issue. I am an infrastructure engineer/SysAdmin that understands code but not a developer. 
This is just nagging me.

**Comment by jooskim** [comment]
@RobTiki The SSL handshake error is complaining about the certificate issued to the VCSA not the file server. @dbarkelew, would you perceive this the same issue as the one you pointed me to before?

**Comment by jooskim** [comment]
@RobTiki It's a bit complicated, there are two different certificates that are verified overall - one from the OVA fileserver and the other from the VCSA itself. What you have done with the OVA thumbprint is perfect, but the error message you originally posted didn't throw out the SSL handshake exception because of the mismatching thumbprint of the OVA appliance, but because of the VCSA's certificate that doesn't work as expected with the VIC UI service. Normally you don't need to worry about this case, but there certainly is a known vCenter Server issue that affected some users including @vmwarelab. And for that I would like to check with @dbarkelew to see if this is the case for you as well. Have you upgraded before your VCSA from 6.0 to 6.5?

Btw, I have attached two different SSL handshake exception error messages below. One was caused by the VCSA certificate and the other was caused by the mismatching thumbprint of the OVA file server.

```
[2017-06-22T15:49:31.655Z] [ERROR] data-service-pool-788 70001518 100020 200002 com.vmware.vic.VicUIPropertyProvider VicUIServiceImpl.getProperties error: com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
```
```
[2017-06-29T16:19:33.344Z] [ERROR] vc-extensionmanager-pool-152 70001104 100490 200003 com.vmware.vise.vim.extension.VcExtensionManager                  Package com.vmware.vic.ui was not installed. Error downloading https://10.17.109.99:9443/com.vmware.vic.ui-v1.1.1.zip. Make sure that the URL is reachable then logout/login to force another download. javax.net.ssl.SSLHandshakeException: com.vmware.vim.vmomi.client.exception.VlsiCertificateException: Server certificate chain is not trusted and thumbprint doesn't match
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)
        at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949)
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302)```

**Comment by RobTiki** [comment]
@jooskim yes, I did upgrade from 6.0 to 6.5 and applied all updates as they were released up to 6.5e. Learning more and working with VMCA it looks like My setup my have deeper issues as I cannot successfully regenerate or reset all my certificates. 

I am going to deploy a new vcsa over the weekend and go from there.

**Comment by danisoltani** [comment]
@jooskim I am having the exact same issue with flex plugin, you can find the exact error message below. A little more about my environment: This is my lab set up, with vcenter 6.5 set up as a intermediate CA. I also have NSX in my lab, and the flex client plugin works there if that matters. HTML5 client plugin for VIC is showing up correctly. I do not see any other error messages in logs regarding certificates and everything else appears to be functioning correctly. 

vCenter version: 6.5.0.5600
VIC version: 1.1.1

[ERROR] data-service-pool-229        70000067 100005 200002 com.vmware.vic.VicUIPropertyProvider                              VicUIServiceImpl.getProperties error: com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match

**Comment by jooskim** [comment]
Hi @RobTiki @danisoltani @vmwarelab. I've been busy with other work recently so my apologies for the delay. I started looking at this issue today and I think I found what might be the very likely cause of the SSL handshake exception that you guys are running into.

To keep things short, the current certificate verification strategy is that it queries for certificates from the current vSphere session where we assume there is only one certificate returned from the session. If there is more than one, however, this would not pass the verification and thus throw an exception.

I have a rough plan to see if I guessed correctly, but I do need an environment where vSphere session returns more than one certificate. So I am asking you guys to do me a favor. What I could do is build the plugin with some debug logging on so you can install the plugin, restart the Flex client, log back in, and send me the Virgo server log file.

Let me know what you think.

**Comment by danisoltani** [comment]
@jooskim Sound's like a plan. I'm down to help at any time.

**Comment by jooskim** [comment]
Here are the plugin zip files with the debug feature on:

https://onevmw-my.sharepoint.com/personal/kjosh_vmware_com/_layouts/15/guestaccess.aspx?docid=085c09d027a964994ba06d8e4c4c738c7&authkey=AaWiKb4vIUQ2Ehn6UMIFV-s&expiration=2017-07-19T05%3a00%3a00.000Z

https://onevmw-my.sharepoint.com/personal/kjosh_vmware_com/_layouts/15/guestaccess.aspx?docid=058c99370fa024dbc8d3dd4b8bccaf06b&authkey=ARPQ4PftTAJt0Xc92137Vzw&expiration=2017-07-19T05%3a00%3a00.000Z

Step 1. Upload the zip files to the OVA appliance
```
scp -r com.vmware.vic*0.0.1.zip root@${IP_TO_OVA_APPLIANCE}:/data/fileserver/files/
```
Step 2. In the folder where you extracted `vic_v1.1.1.tar.gz` update `ui/plugin-manifest` file to look like the following:
```
key_flex="com.vmware.vic.ui"
name="VicUI"
version="0.0.1"
summary="VicUI"
company="VMware"
key_h5c="com.vmware.vic"
```
Step 3. Navigate to `ui/VCSA` and run `upgrade.sh`
Step 4. Once done, SSH into the VCSA and restart the Flex Client and H5 Client services by typing the following commands:
```
ssh root@${IP_TO_VCSA}
service-control --stop vsphere-client && service-control --start vsphere-client && service-control --stop vsphere-ui && service-control --start vsphere-ui
exit
```
Step 5. Log into the Flex Client and navigate to any VCH appliance in the "Hosts and Clusters" view you have deployed previously
Step 6. Once page load is complete, type the following command to download the Virgo server log file onto your computer:
```
scp /var/log/vmware/vsphere-client/logs/vsphere_client_virgo.log ${TARGET_LOCATION}
```
Step 7. Email the log file to kjosh@vmware.com

Thanks very much for helping me troubleshoot this long-lived issue. Your help is much appreciated :)

**Comment by jooskim** [comment]
Some of the log messages you should see are as follows:
![image](https://user-images.githubusercontent.com/3834071/28141212-3a0700c0-6721-11e7-8895-2740539ca531.png)
![image](https://user-images.githubusercontent.com/3834071/28141236-51e95364-6721-11e7-8571-6d97698df014.png)

**Comment by danisoltani** [comment]
Done, sent.

**Comment by jooskim** [comment]
Closing this issue for the sake of clearing high priority issues for 1.2. @danisoltani @RobTiki @vmwarelab I know this issue is not solved yet, but I'm still available for troubleshooting this issue. Please refer to my last comment above for troubleshoot instructions and send me the log file at your best convenience. Thanks.

**Comment by mdubya66** [comment]
Reopened as it's not fixed. Can we get the requested information as soon as possible.

**Comment by jooskim** [comment]
I installed both the Flex and Client plugins on a linked VC environment that consists of two 6.5.0d VCSA instances where I hoped to verify if multiple VC certificates were the primary reason for the SSL handshake error in the Flex Client, but unfortunately I could not reproduce the issue with the latest OVA build 13338; I did not see any thumbprint related exception in the Virgo server log and the portlets would show up as expected no matter what VC's Flex Client I was logged into.

The Flex Client plugin that shipped in the OVA release version 1.1.1 (56a309fb) would still fail to show up in the Flex Client, not because of the certificate issue, but because of another issue which already was addressed and will be shipping with 1.2 release.

In other words, this issue (specifically involving thumbprint exceptions in vic-service) is not reproducible in my every possible scenario, and is thus likely to be resulting from external factors such as the vCenter bug as seen in https://ikb.vmware.com/kb/2109074. Upgrading to version 1.2 should bring back the Flex Client plugin for most cases.
